index.js

```diff
@@ -0,0 +1,217 @@
+//. # Sanctuary Benchmark
+//.
+//. Small wrapper over [benchmarkjs][1] to enable a consistent style of
+//. benchmarking across all Sanctuary projects. Allows for the easy definition
+//. of comparative benchmarks, and outputs results in a standardized format for
+//. sharing.
+//.
+//. ## Usage
+//.
+//. ```js
+//. //
+//. // old-vs-new.js
+//. //
+//.
+//. //Imagine these are libs. Normally they would be require()'d.
+//. const oldVersion = (f, xs) => xs.map(f);
+//. const newVersion = (f, xs) => {
+//.   const len = xs.length; out = new Array(len);
+//.   for(let idx = 0; idx < len; idx++) out[idx] = f(xs[idx]);
+//.   return out;
+//. };
+//.
+//. const small = Array.from({length: 1}, (_, i) => i);
+//. const large = Array.from({length: 1000}, (_, i) => i);
+//.
+//. //Normally we wouldn't assign this to a var, but just export it.
+//. const benchmark = require('sanctuary-benchmark')(oldVersion, newVersion, {
+//.   'map/small': map => map(x => x + 1, small),
+//.   'map/large': map => map(x => x + 1, large)
+//. });
+//.
+//. //
+//. // index.js
+//. //
+//.
+//. //Here we would map command line arguments to benchmarking options. We
+//. //would also determine which benchmarks to require(). In this case I've
+//. //hard-coded the old-vs-new.js file, but we might just as well run a
+//. //ramda-vs-sanctuary.js suite.
+//. benchmark({
+//.   colors: true,
+//.   leftHeader: 'declarative',
+//.   match: '**',
+//.   minSamples: 10,
+//.   rightHeader: 'imperative',
+//.   significantChange: 5
+//. });
+//. ```
+'use strict';
+
+var benchmark = require('benchmark');
+var Table = require('cli-table2');
+var micromatch = require('micromatch');
+
+
+//       formatPct :: Number -> String
+function formatPct(pct) {
+  var rendered = pct.toLocaleString('en', {
+    minimumIntegerDigits: 3,
+    minimumFractionDigits: 1,
+    maximumFractionDigits: 1,
+    useGrouping: false
+  });
+  return (pct < 0 ? rendered : '+' + rendered) + '%';
+}
+
+//       format :: BenchmarkResult -> String
+function format(res) {
+  return res.hz.toLocaleString('en', {maximumFractionDigits: 0}) + ' Hz '
+       + '±' + res.stats.rme.toFixed(2) + '% '
+       + '(n ' + String(res.stats.sample.length) + ')';
+}
+
+//       get :: (String, a, Any) -> a
+function get(key, fallback, x) {
+  var o = Object(x);
+  return typeof o[key] === typeof fallback ? o[key] : fallback;
+}
+
+//       green :: String -> String
+function green(s) {
+  return '\u001B[32m' + s + '\u001B[0m';
+}
+
+//       identity :: a -> a
+function identity(x) {
+  return x;
+}
+
+//       red :: String -> String
+function red(s) {
+  return '\u001B[31m' + s + '\u001B[0m';
+}
+
+//.
+//. ## API Documentation
+//.
+//. <details><summary>
+//. <code>default :: (a, b, StrMap (Spec a b)) -> Options -> Undefined</code>
```

Comment:
Why `($1, $2, $3) -> $4 -> $5` rather than `$1 -> $2 -> $3 -> $4 -> $5`? Are you concerned that `S.curry4` would add noticeable overhead even with type checking disabled?